### ISSUE TITLE
[Bug] GNN training ignores requested learning rate

### ISSUE THREAD

**Issue by ThunderKhan** [KanishJebaMathewM/Truxify#15788]
**Describe the bug**

The GNN training API accepts a `learning_rate` parameter, but the requested value is not propagated to the optimizer's training routine.

`TrainRequest` defines `learning_rate` as a configurable request parameter, while `RouteOptimizer.train()` currently creates the Adam optimizer with a hard-coded learning rate of `0.001`. As a result, callers cannot control the training learning rate through the API even when they provide a different value.

This creates a mismatch between the API contract and the actual training behavior.

**To Reproduce**

1. Send a request to the GNN training endpoint with a custom learning rate, for example:

   ```json
   {
     "epochs": 10,
     "learning_rate": 0.01
   }
   ```
2. Observe that the training endpoint accepts the value.
3. Trace the call into `RouteOptimizer.train()`.
4. Observe that the Adam optimizer is initialized with the hard-coded value `0.001` instead of the requested `0.01`.
5. Repeat with another learning rate and observe that the optimizer configuration remains unchanged.

The relevant flow is:

```text
TrainRequest.learning_rate
        ↓
train_model()
        ↓
RouteOptimizer.train(...)
        ↓
Adam(..., lr=0.001)
```

**Expected behavior**

The learning rate supplied by the caller should be propagated from the API request to `RouteOptimizer.train()` and used when constructing the training optimizer.

For example, a request using:

```json
{
  "learning_rate": 0.01
}
```

should result in the underlying Adam optimizer being configured with:

```python
lr=0.01
```

When no custom learning rate is supplied, the existing default value should be preserved.

The change should remain backward-compatible and should not alter other training behavior.

**Screenshots**

Not applicable. This is a backend ML training configuration issue.

**Environment (please complete the following information):**

* OS: Windows 11
* Python: 3.12
* Component: `backend/ml/gnn/models.py`
* API component: `backend/ml/routes/gnn_routes.py`
* Affected flow: GNN model training
* Test framework: `pytest`

**Additional context**

The API already exposes `learning_rate` through `TrainRequest`, but the value is discarded at the boundary before reaching the training implementation. The underlying training method also defines its own default learning rate.

The fix should:

* accept the requested learning rate in `RouteOptimizer.train()`;
* use that value when constructing the Adam optimizer;
* forward the API request value from `train_model()`;
* preserve the current default when the caller omits the parameter;
* add regression coverage for both custom and default learning-rate behavior.

This issue concerns parameter propagation and training configuration only; it does not require changing the GNN architecture or optimization algorithm.

**Comment by ThunderKhan** [comment]
I’m claiming this issue.

I’ve traced the defect through the GNN training path and confirmed that the API accepts `learning_rate`, but the value is discarded before optimizer construction, where `RouteOptimizer.train()` currently uses a hard-coded rate.

I’m implementing a focused fix to propagate the requested learning rate through the API and training layer, while preserving the existing default behavior. I’m also adding regression tests covering both custom and default learning-rate handling.

I’m opening the PR in a few minutes.
